### PR TITLE
Updated Portal.js

### PR DESCRIPTION
**Pull request by FlameSoulis**
This updates portal.js to the one used currently in world while also addressing Bug 3374, regarding clicking back on GoTo and being teleported again (as the teleporter was the last location the avatar was upon departure).

Testing conditions:
1. Set up teleporter as usual (JSON area should contain domain name of destination)
2. Install script
3. Walk into teleporter. Teleporter should take user to desination
4. Open up GoTo.
5. Click the back button. User should return to previous area inside teleporter.
6. With current update from this pull, user should **NOT** be teleported. Original verison would teleport user back to destination.

### CODE REVIEW

File: DomainContent/Home/portal.js [hunk 1]
```diff
@@ -1,52 +1,93 @@
...
-            }
+    this.enterEntity = function (entityID) {
+        // check if we should teleport
+        if (canTeleport === false) {
+            // if we have not passed enough time, do not teleport the user
+            return;
         }
+        // get latest props in case we changed the destination
+        getProps(entityID);
+        print("enterEntity() .... The portal destination is " + portalDestination);
```
Comment (elisa-lj11) [inline DomainContent/Home/portal.js]: remove extra print statements

File: DomainContent/Home/portal.js [hunk 2]
```diff
@@ -1,52 +1,93 @@
...
+
+        if (portalDestination.length > 0) {
+            print("Teleporting to hifi://" + portalDestination);
```
Comment (elisa-lj11) [inline DomainContent/Home/portal.js]: remove extra print statements

File: DomainContent/Home/portal.js [hunk 3]
```diff
@@ -1,52 +1,93 @@
...
+            Window.location = "hifi://" + portalDestination;
+        }
+
+    };
+
+    this.leaveEntity = function (entityID) {
+        print("leaveEntity() called ....");
```
Comment (elisa-lj11) [inline DomainContent/Home/portal.js]: remove extra print statements

File: DomainContent/Home/portal.js [hunk 4]
```diff
@@ -1,52 +1,93 @@
...
+            // only play the sound if we can teleport
+            playSound(entityID);
+        }
+        canTeleport = true;
     };
 
-    this.leaveEntity = function(entityID) {
-        print("portal.js | leaveEntity");
+    this.hoverEnterEntity = function (entityID) {
+        print("hoverEnterEntity() called ....");
```
Comment (elisa-lj11) [inline DomainContent/Home/portal.js]: remove extra print statements

File: DomainContent/Home/portal.js [hunk 5]
```diff
@@ -1,52 +1,93 @@
...
+        if (!animationURL) {
+            getProps(entityID);
+        }
+        Entities.editEntity(entityID, {
+            animation: { url: animationURL, fps: 24, firstFrame: 1, lastFrame: 25, currentFrame: 1, running: true, hold: true }
```
Comment (elisa-lj11) [inline DomainContent/Home/portal.js]: use constants instead of hard code

File: DomainContent/Home/portal.js [hunk 6]
```diff
@@ -1,52 +1,93 @@
...
-        var properties = Entities.getEntityProperties(thisEntityID, 'position');
-        if (properties) {
-            Audio.playSound(teleport, { position: properties.position, volume: 0.40, localOnly: true });
+    function playSound(entityID) {
+        print("playing teleport sound");
+        if (teleportSound.downloaded) {
+            if (!position) {
+                getProps(entityID);
+            }
+            Audio.playSound(teleportSound, { position: position, volume: 0.40, localOnly: true });
```
Comment (elisa-lj11) [inline DomainContent/Home/portal.js]: use a constant for volume instead of hard code

**Comment by FlameSoulis** [comment]
Okay, I am slowly going into Visual Studio Code and updating the project's settings for the format system (namely adding "javascript.format.insertSpaceAfterFunctionKeywordForAnonymousFunctions": false to the workspace settings after updating the git exclude file).
I will take a look at the print statements (again, not my code) but for anything like "use a constant for volume instead of hard code", again, that wasn't my code, and my JavaScript knowledge isn't the best (primarily a C++ coder).

**Comment by FlameSoulis** [comment]
Okay, there we go! Cleaned up the script a bit (ignore the eslintrc.js commits, accidentally merged everything to the master which made things tricky for PR103). Removed all print statements (except the one regarding loading the script), made the volume a constant, and removed all extra events and calls to animations, as none of the portals in use have an animation active. If, however, you feel this should be added back, let me know and I will address this right away.

**Comment by misslivirose** [comment]
@FlameSoulis can you write out more detail in your test plan to include the way you'd expect the portal json setup and what you mean by "install script"?

**Comment by FlameSoulis** [comment]
The JSON is still the same as the older portals: Just directly add in the address (Example: "hifi:/painting") as though it was a text entry (no normal JSON conventions, that was the way the script has been since creation). As for the "install script", I meant by adding it to the Script URL of an entity.

Whatever test plans used for the original portal script should completely apply to this one, as none of the ways it operates changes except for the protection of re-entry.

**Comment by misslivirose** [comment]
We probably don't have an existing test plan for portal.js. I will file a case with QA to create one.

**Comment by misslivirose** [comment]
Alan confirmed this is looking good for the new portals, so merging.

**Pull request merged**